plugin.go

```diff
@@ -150,46 +109,69 @@ func (p Plugin) Exec() error {
 			body, err := ioutil.ReadAll(resp.Body)
 
 			if err != nil {
-				fmt.Printf("Error: Failed to read the HTTP response body. %s\n", err)
+				log.WithFields(log.Fields{
+					"error": err,
+				}).Error("Failed to parse HTTP response")
 			}
 
-			if p.Config.Debug {
-				fmt.Printf(
-					debugFormat,
-					i+1,
-					req.URL,
-					req.Method,
-					req.Header,
-					string(b),
-					resp.Status,
-					string(body),
-				)
-			} else {
-				fmt.Printf(
-					respFormat,
-					i+1,
-					req.URL,
-					resp.Status,
-					string(body),
-				)
+			output, err := template.RenderTrim(result, webhook{
+				Debug:    p.Config.Debug,
+				Number:   i,
+				URL:      req.URL.String(),
+				Method:   req.Method,
+				Header:   req.Header,
+				Status:   resp.Status,
+				Request:  string(b),
+				Response: string(body),
+			})
+
+			if err != nil {
+				log.WithFields(log.Fields{
+					"error": err,
+				}).Error("Failed to parse debug template")
+
+				return err
 			}
+
+			fmt.Println(output)
 		}
 
 		if len(p.Config.ValidCodes) > 0 && !intInSlice(p.Config.ValidCodes, resp.StatusCode) {
-			return fmt.Errorf("Error: Response code %d not found among valid response codes", resp.StatusCode)
-		}
+			log.WithFields(log.Fields{
+				"code": resp.StatusCode,
+			}).Error("Valid response code not found")
 
+			return fmt.Errorf("valid response code not found")
+		}
 	}
 
 	return nil
 }
 
-// Function checks if int is in slice of ints
-func intInSlice(s []int, e int) bool {
-	for _, a := range s {
-		if a == e {
-			return true
+func (p Plugin) payload() ([]byte, error) {
+	if p.Config.Template == "" {
+		res, err := json.Marshal(&p)
+
+		if err != nil {
+			log.WithFields(log.Fields{
```

Comment:
logrus has a `WithError(err)` that you can use instead of manually defining fields. Also when there is only a single field you can use `WithField(key, value)`